src/perf_counters.h

```diff
@@ -90,10 +90,13 @@ class BENCHMARK_EXPORT PerfCounters final {
   // True iff this platform supports performance counters.
   static const bool kSupported;
 
-  bool IsValid() const { return !counter_names_.empty(); }
+  bool IsValid() const { return counter_ids_.size() == counter_names_.size(); }
```

Comment:
is this ever not the case?

Reply:
To be fair, this was just me trying to justify the existence of `IsValid()` at this point.

Reply:
if we don't need the IsValid we can drop it.